wgpu-core/src/instance.rs

```diff
@@ -685,20 +678,16 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                 .queue_families
                 .iter()
                 .find(|family| family.queue_type().supports_graphics())
-                .unwrap();
-            let mut gpu = unsafe { phd.open(&[(family, &[1.0])], enabled_features).unwrap() };
+                .ok_or(RequestDeviceError::NoGraphicsQueue)?;
+            let mut gpu = unsafe { phd.open(&[(family, &[1.0])], enabled_features)? };
 
             let limits = phd.limits();
-            assert_eq!(
-                0,
-                BIND_BUFFER_ALIGNMENT % limits.min_storage_buffer_offset_alignment,
-                "Adapter storage buffer offset alignment not compatible with WGPU"
-            );
-            assert_eq!(
-                0,
-                BIND_BUFFER_ALIGNMENT % limits.min_uniform_buffer_offset_alignment,
-                "Adapter uniform buffer offset alignment not compatible with WGPU"
-            );
+            if BIND_BUFFER_ALIGNMENT % limits.min_storage_buffer_offset_alignment != 0 {
+                return Err(RequestDeviceError::IncompatibleStorageBufferOffsetAlignment);
+            }
+            if BIND_BUFFER_ALIGNMENT % limits.min_uniform_buffer_offset_alignment != 0 {
+                return Err(RequestDeviceError::IncompatibleUniformBufferOffsetAlignment);
+            }
```

Comment:
I'm not sure if it's worth keeping two different variants for these types, or if I should merge them into a single generic one.

My reasoning is: the user can't do much anyway if their device doesn't support WebGPU alignments.

Reply:
actually, just turn them back into assertions. It's *not* expected to happen, it means the specification is wrong.